Mathlib/Tactic/FindSyntax.lean

```diff
@@ -75,12 +75,20 @@ elab "#find_syntax " id:str d:(&" approx")? : command => do
       if !declName.isInternal && !ls.isEmpty then symbs := symbs.insert (declName, ls)
   -- From among the parsers in `symbs`, we extract the ones whose `symbols` contain the input `str`
   let mut msgs := #[]
+  let env ← getEnv
+  let allModules := env.allImportedModuleNames
   for (nm, ar) in symbs.toList do
     let rem : String := " _ ".intercalate (ar.map litToString).toList
     -- If either the name of the parser or the regenerated syntax stub contains the input string,
     -- then we include an entry into the final message.
     if 2 ≤ (nm.toString.splitOn id.getString).length || 2 ≤ (rem.splitOn id.getString).length then
-      msgs := msgs.push <| .ofConstName nm ++ m!":\n  '{rem.trim}'\n"
+      let some mod := env.getModuleIdxFor? nm | pure default
+      let some rgs ← findDeclarationRanges? nm | pure default
+      let rg := rgs.range
+      msgs := msgs.push <| .ofConstName nm ++ m!": \
+        defined in '{allModules[mod.toNat]!}'\n\
```

Comment:
I'm a bit confused how this actually works: `mod` is a `ModuleIdx`, `mod.toNat` a natural number, while `allModules` is just an Array of names. Is it guaranteed that these match up? Can you submit a PR documenting this? I feel slightly uneasy relying on undocumented internals, if these are unstable.

Sorry for putting that work on you; I know you're just using what you've got. But I'd rather gradually make this system more robust.

Reply:
I asked during office hours that this was indeed the case and it was confirmed that yes, this is a current feature and one that is unlikely to change.  [lean4#6749](https://github.com/leanprover/lean4/pull/6749)

Reply:
Thanks for checking!